### PR TITLE
Add test to check basic interactions with layer controls widgets

### PR DESCRIPTION
**Pull request by dalthviz**
# Fixes/Closes

Initial effort towards improving GUI testing automation: https://napari.zulipchat.com/#narrow/stream/212875-general/topic/GUI.20testing

# Description

This is an initial attempt to add more tests for code related with GUI elements. The new parametrized test proposed creates an instance of a layer (`Image`, `Labels`, `Points`, `Shapes`, `Surface`, `Tracks` or `Vectors`) and then operates over the layer control widgets simulating user usage (`QCombox`, `QSpinBox` and `QSlider` based widgets value changes, clicking `QCheckbox` and buttons widgets, `QColorSwatch` widgets color changes typing colors, etc)

This also fixes a little issue that I noticed while working on the test. The error is related with the value of the color mode combobox for a Vectors layer controls. The combobox was not showing the correct value after a fallback to a valid value is done (when a `ValueError` exception is raised, the color mode value of the layer is changed to the previous one, but the combobox value was missing being updated too):

* Before:

vectors_before

* After:

vectors_after

Besides that, the test structure also has some logic to skip some tracebacks from issues which are still open/were discovered while writing the test like #5798 and #4864

# References

## Type of change

- [x] Add more tests

# How has this been tested?

- [x] all tests pass with my change
- [x] I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by alisterburt — COMMENTED** [review]
Great stuff @dalthviz - thank you

Would love to see this in but the test failures on pyside look real
https://github.com/napari/napari/actions/runs/4768301656/jobs/8477468793?pr=5757

Any ideas how we could get around those?

**Comment by dalthviz** [comment]
Hi @alisterburt, thank you for giving this a check! I think the problem comes from PySide2/6 not having the `findChildren` signature that supports passing a tuple of types (which seems like is only available for PyQt 5/6 bindings). Will push a fix for that :+1:

**Comment by kephale** [comment]
Looks great! 

@kne42 any thoughts on this one? In particular, are there some other parts of the GUI you were thinking needed more automated testing?

File: napari/_qt/layer_controls/_tests/test_qt_layer_controls.py [hunk 1]
```diff
@@ -1,31 +1,333 @@
...
+                value_range = np.append(value_range, qspinbox_max - 1)
+        for value in value_range:
+            qspinbox.setValue(value)
+            # capture any output done to sys.stdout or sys.stderr.
+            captured = capsys.readouterr()
+            assert not captured.out
+            if captured.err:
+                # since an error was found check if it is associated with a known issue still open
+                expected_errors = [
+                    "MemoryError: Unable to allocate",  # See https://github.com/napari/napari/issues/5798
```
Comment (GenevieveBuckley) [inline napari/_qt/layer_controls/_tests/test_qt_layer_controls.py:216]: Can you make a note in each of these issues that any PR to close them should also remove these lines from this test? (You might have to do it after this PR is merged, in order to link to the line(s) directly)

File: napari/_qt/layer_controls/_tests/test_qt_layer_controls.py [hunk 2]
```diff
@@ -1,31 +1,333 @@
...
+                    "ValueError: array is too big; `arr.size * arr.dtype.itemsize` is larger than the maximum possible size.",  # See https://github.com/napari/napari/issues/5798
+                    "ValueError: Maximum allowed dimension exceeded",  # See https://github.com/napari/napari/issues/5798
+                    "IndexError: index ",  # See https://github.com/napari/napari/issues/4864
+                    "RuntimeWarning: overflow encountered",  # See https://github.com/napari/napari/issues/4864
+                ]
```
Comment (GenevieveBuckley) [inline napari/_qt/layer_controls/_tests/test_qt_layer_controls.py:221]: Can you make a note in each of these issues that any PR to close them should also remove these lines from this test? (You might have to do it after this PR is merged, in order to link to the line(s) directly)

**Review by GenevieveBuckley — APPROVED** [review]

File: napari/_qt/layer_controls/_tests/test_qt_layer_controls.py (hunk 2 above)
Comment (dalthviz) [reply napari/_qt/layer_controls/_tests/test_qt_layer_controls.py:221]: Thanks for the feedback! Will do :+1:

**Comment by GenevieveBuckley** [comment]
Nice work! Thank you @dalthviz 🎉 

If you can follow up on [this suggestion](https://github.com/napari/napari/pull/5757#discussion_r1189319525) that would be fantastic.

**Pull request merged**